nomad/structs/structs.go

```diff
@@ -1474,14 +1490,21 @@ type Allocation struct {
 	ModifyIndex uint64
 }
 
-// TerminalStatus returns if the desired status is terminal and
-// will no longer transition. This is not based on the current client status.
+// TerminalStatus returns if the desired or actual status is terminal and
+// will no longer transition.
 func (a *Allocation) TerminalStatus() bool {
 	switch a.DesiredStatus {
 	case AllocDesiredStatusStop, AllocDesiredStatusEvict, AllocDesiredStatusFailed:
 		return true
 	default:
-		return false
+		// If all tasks are dead, the alloc is terminal.
```

Comment:
Do we need this logic here? I think ideally the client is going to do this and set the Alloc Client status.